homepage/homepage/app/docs/[framework]/[...slug]/schemas/accounts-and-migrations.mdx

```diff
@@ -0,0 +1,181 @@
+  root = co.ref(MyAppRoot);
+}
+```
+</CodeGroup>
+If you want to extend the `profile` to contain additional fields (such as an avatar `ImageDefinition`), you can declare your own profile schema class that extends `Profile`:
+
+<CodeGroup>
+{/* prettier-ignore */}
+```ts
+import { Account, Profile, ImageDefinition } from "jazz-tools";
+
+export class MyAppAccount extends Account {
+  profile = co.ref(MyAppProfile);
+  root = co.ref(MyAppRoot);// old
+}
+
+export class MyAppRoot extends CoMap {// old
+  myChats = co.ref(ListOfChats);// old
+  myContacts = co.ref(ListOfAccounts);// old
+}// old
+
+export class MyAppProfile extends Profile {
+  name = co.string; // compatible with default Profile schema
+  avatar = co.ref(ImageDefinition);
+}
+```
+</CodeGroup>
+
+## Resolving CoValues starting at `profile` or `root`
+
+<ContentByFramework framework="react">
+To use per-user data in your app, you typically use `useAccount` somewhere in a high-level component, specifying which references to resolve using a depth-spec (see [Subscribing & deep loading](/docs/using-covalues/subscription-and-loading)).
+
+<CodeGroup>
+{/* prettier-ignore */}
+```tsx
+import { useAccount } from "jazz-react";
+
+function DashboardPageComponent() {
+  const { me } = useAccount({ profile: {}, root: { myChats: {}, myContacts: {}}});
+
+  return <div>
+    <h1>Dashboard</h1>
+    {me ? <div>
+      <p>Logged in as {me.profile.name}</p>
+      <h2>My chats</h2>
+      {me.root.myChats.map((chat) => <ChatPreview key={chat.id} chat={chat} />)}
+      <h2>My contacts</h2>
+      {me.root.myContacts.map((contact) => <ContactPreview key={contact.id} contact={contact} />)}
+    </div> : "Loading..."}
+  </div>
+}
+
+```
+</CodeGroup>
+</ContentByFramework>
+
+
+## Populating and evolving `root` and `profile` schemas with migrations
+
+As you develop your app, you'll likely want to
+
+- initialise data in a user's `root` and `profile`
+- add more data to your `root` and `profile` schemas
+
+You can achieve both by overriding the static `migrate()` method on your `Account` schema class.
+
+### When migrations run
+
+Migrations are run after account creation and every time a user logs in.
+Jazz waits for the migration to finish before passing the account to your app's context.
+
+### Initialising user data after account creation
+
+<CodeGroup>
+{/* prettier-ignore */}
+```ts
+export class MyAppAccount extends Account {
+  static async migrate(
+    me: MyAppAccount,
+    creationProps: { name: string } // from AuthMethod
+  ) {
+    // we specifically need to check for undefined, because the profile might simply be not loaded (`null`) yet
+    if (me.profile === undefined) {
+      // this is the default profile setup migration and if you don't need anything else, you can leave out this whole block
+      const publiclyReadableGroup = Group.create({ everyone: "reader" });
+      me.profile = Profile.create({ name: creationProps.name }, { owner: publiclyReadableGroup });
+    }
```

Comment:
Love this new approach, but won't work now.

Going to update the example to something that works.

Added an issue to track the idea: https://github.com/garden-co/jazz/issues/1161